cstate/powermgr_cstate_tests.sh

```diff
@@ -0,0 +1,322 @@
+    turbostat_CC6_value=$(echo "scale=2; $CC6_val > 0.00" | bc)
+    if [[ $turbostat_CC6_value -eq 1 ]]; then
+        test_print_trc "CPU Core C6 residency is available \
+during idle when $current_cpuidle_driver is running"
+    else
+        die "Did not get CPU Core C6 residency during idle \
+when $current_cpuidle_driver is running"
+    fi
+}
+
+test_cpu_core_c7_residency_intel_idle() {
+    check_intel_idle
+    judge_cc7_residency_during_idle
+}
+
+test_cpu_core_c7_residency_intel_s2idle() {
+    check_intel_idle
+    judge_cc7_residency_during_s2idle
+}
+
+test_cpu_core_c6_residency_intel_idle() {
+    check_intel_idle
+    judge_cc6_residency_during_idle
+}
+
+cc_state_disable_enable() {
+    local cc=$1
+    local setting=$2
+
+    for ((i = 0; i < cpu_num; i++)); do
+        #Find Core Cx state
+        cc_num=$(grep . /sys/devices/system/cpu/cpu0/cpuidle/state*/name |
+            sed -n "/$cc$/p" | awk -F "/" '{print $8}' | cut -c 6)
+        test_print_trc "Core $cc state name is: $cc_num"
+        [[ -n "$cc_num" ]] || block_test "Did not get Core $cc state."
+        #Change Core Cx state
+        do_cmd "echo $setting > /sys/devices/system/cpu/cpu$i/cpuidle/state$cc_num/disable"
+        let deeper=$cc_num+1
+        #Change deeper Core Cx state
+        for ((j = deeper; j < state_num; j++)); do
+            do_cmd "echo $setting > /sys/devices/system/cpu/cpu$i/cpuidle/state$j/disable"
+        done
+    done
+}
+
+disable_cc_check_pc() {
+    local cc=$1
+    local pc_y=$2
+    local pc_n=$3
+    local cpu_num
+    local columns
+
+    cpu_num=$(lscpu | grep "^CPU(s)" | awk '{print $2}')
+    state_num=$(ls "${CPU_BUS_SYSFS_PATH}"/cpu0/cpuidle | grep -c state)
+    columns="Pkg%pc2,Pkg%pc3,Pkg%pc6,Pkg%pc7,Pkg%pc8,Pkg%pc9,Pk%pc10"
+
+    cc_state_disable_enable "$cc" 1
+
+    #Check Package Cstates, CC10 disable--> expect PC8 only
+    #CC8 and deeper disable--> PC6 only
+    tc_out=$(turbostat -q --show $columns -i 1 sleep 20 2>&1)
+    [[ -n "$tc_out" ]] || die "Did not get turbostat log"
+    test_print_trc "turbostat tool output: $tc_out"
+    pc_y_res=$(echo "$tc_out" |
+        awk '{for(k=0;++k<=NF;)a[k]=a[k]?a[k] FS $k:$k} END{for(k=0;k++<NF;)print a[k]}' |
+        grep "$pc_y" | awk -F " " '{print $3}')
+    pc_n_res=$(echo "$tc_out" |
+        awk '{for(k=0;++k<=NF;)a[k]=a[k]?a[k] FS $k:$k} END{for(k=0;k++<NF;)print a[k]}' |
+        grep "$pc_n" | awk -F " " '{print $3}')
+    [[ -n "$pc_y_res" ]] || die "Did not get $pc_y state."
+    [[ -n "$pc_n_res" ]] || die "Did not get $pc_n state."
+    if [[ $(echo "scale=2; $pc_y_res > 0.00" | bc) -eq 1 ]] && [[ $pc_n_res == "0.00" ]]; then
+        cc_state_disable_enable "$cc" 0
+        test_print_trc "Expected to get $pc_y only when disable $cc and deeper state.\
+$pc_y residency: $pc_y_res; $pc_n residency: $pc_n_res"
+    else
+        cc_state_disable_enable "$cc" 0
+        die "Did not get $pc_y residency after disable $cc and deeper states. \
```

Comment:
Did you really mean a 'die' when the test is failed, or just a 'return', which makes the script keep running the reset tests?

Reply:
Will not change "die" usage.